### PR TITLE
Update free-courses-en.md

### PR DESCRIPTION
**Pull request by Imbiber**
## What does this PR do?
Add resource(s) 

## For resources
Lecture for Theory of Computation by Prof. Subrahmanyam Kalyanasundaram (IIT-Hyderabad)
### How do we know it's really free?
Youtube playlist

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -1604,7 +1604,9 @@
 * [Automata Theory](https://online.stanford.edu/courses/soe-ycsautomata-automata-theory)
 * [Formal Languages & Automata Theory](https://www.youtube.com/playlist?list=PLLvKknWU7N4zvTGcw9N2_7eZSTTkryb0U) - Lalit Vashistha
 * [Intro to Theoretical Computer Science](https://www.udacity.com/course/intro-to-theoretical-computer-science--cs313) (Udacity)
+* [Theory of Computation](https://www.youtube.com/playlist?list=PLyqSpQzTE6M9-v4V62bCygAVYGluaijyo) - Prof. Subrahmanyam Kalyanasundaram
 * [Theory of Computation & Automata Theory](https://www.youtube.com/playlist?list=PLBlnK6fEyqRgp46KUv4ZY69yXmpwKOIev) - Neso Academy
+
```
Comment (Thenlie) [inline courses/free-courses-en.md:1609]:
```suggestion
```

**Comment by Thenlie** [comment]
The link looks good to me, so should be ready once the extra line is removed.

**Comment by Imbiber** [comment]
> The link looks good to me, so should be ready once the extra line is removed.

Done

**Review by Thenlie — APPROVED** [review]

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -1604,6 +1604,7 @@
 * [Automata Theory](https://online.stanford.edu/courses/soe-ycsautomata-automata-theory)
 * [Formal Languages & Automata Theory](https://www.youtube.com/playlist?list=PLLvKknWU7N4zvTGcw9N2_7eZSTTkryb0U) - Lalit Vashistha
 * [Intro to Theoretical Computer Science](https://www.udacity.com/course/intro-to-theoretical-computer-science--cs313) (Udacity)
+* [Theory of Computation](https://www.youtube.com/playlist?list=PLyqSpQzTE6M9-v4V62bCygAVYGluaijyo) - Prof. Subrahmanyam Kalyanasundaram
```
[the remark this answers is not in the document]
Comment (Imbiber) [reply courses/free-courses-en.md]: Done.

**Pull request merged**